source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -637,38 +637,35 @@ void GlslShaderGenerator::emitVariableDeclaration(const ShaderPort* variable, co
                                                   GenContext&, ShaderStage& stage,
                                                   bool assignValue) const
 {
-    // A file texture input needs special handling on GLSL
     if (variable->getType() == Type::FILENAME)
     {
-        // Samplers must always be uniforms
```

Comment:
Samplers must actually be either uniforms or function arguments, and `GlslFragmentGenerator` calls this function to do the latter. Default value initializers on the other hand are not permitted for samplers because they are file paths.